src/opentimelineio/composition.h

```diff
@@ -23,7 +23,7 @@ class Composition : public Item {
 
     virtual std::string const& composition_kind() const;
 
-    std::vector<Retainer<Composable>> const& children() const {
+    std::vector<Retainer<Composable>> const& children() const noexcept {
```

Comment:
return by ref is scary here, but I don't want to mess with it just now. The vector is const, but the Composables can be mutated, not what the intent of the const here.

Reply:
Out of curiosity why not change it if we're removing the other ```const&```?

Reply:
the intent of const& here is that if we get children we cannot mutate them, so we wouldn't get the intended safety by removing the const&.

 In other words, children() should really be returning something like `std::vector<Immutable<Composable>>` but we haven't got an Immutable Retainer thingy or a ConstRetainer, or a Retainer<const T>, as far as I know. So making that change seems out of scope to me for this change, as we'd first need to figure out what the returned vector should really be, possibly write a new class for it, then patch it in everywhere appropriate.

Reply:
In this particular case I don't think the ```const&``` was meant to imply immutable Composables, but rather as an optimization/pessimization to avoid copying the list. For example:
https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/835fc5c12791dd29c333e5c6d7e633565b2d792f/src/opentimelineio/timeline.cpp#L35

Reply:
Hmm, what about this:
https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/835fc5c12791dd29c333e5c6d7e633565b2d792f/src/opentimelineio/serializableObjectWithMetadata.h#L28

Replace with ```set_metadata```?

Reply:
Since all we are doing is exposing a member with full access, it seems like the member should be public, and the accessors omitted....

Reply:
I agree with your point, but not only should the children list be immutable, but c in the `auto c` would itself should be const. I think a reasonable interim would be to actually return a copy of the vector, since the children are reference counted...

Reply:
I personally like the accessors since it gives us the ability to add extra functionality later, and seems similar to the rest of the API. My vote would be for ```set_metadata()```.

Reply:
I think I agree? I'd prefer

```cpp
void set_metadata(const AnyDictionary&);
void merge_metadata(const AnyDictionary&);
```
These aren't accessors, but the idea that well-meaning arbitrary end users can reach in and tickle the internal state gives me the shivers. I prefer a functional approach for this kind of thing.

Reply:
In order to unblock and land this work, I'd like to defer this discussion about accessors to another PR or Issue.